core/src/test/java/de/danoeh/antennapod/core/feed/LocalFeedUpdaterTest.java

```diff
@@ -0,0 +1,258 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.robolectric.Shadows.shadowOf;
+
+/**
+ * Test local feeds handling in class LocalFeedUpdater.
+ */
+@RunWith(RobolectricTestRunner.class)
+public class LocalFeedUpdaterTest {
+
+    /**
+     * URL to locate the local feed media files on the external storage (SD card).
+     * The exact URL doesn't matter here as access to external storage is mocked
+     * (seems not to be supported by Robolectric).
+     */
+    private static final String FEED_URL =
+            "content://com.android.externalstorage.documents/tree/primary%3ADownload%2Flocal-feed";
+
+    private Context context;
+    private File localFeedDir1;
+    private File localFeedDir2;
+
+    @Before
+    public void setUp() {
+
+        // Initialize environment
+        context = InstrumentationRegistry.getInstrumentation().getContext();
+        UserPreferences.init(context);
+
+        // Initialize database
+        PodDBAdapter.init(context);
+        PodDBAdapter.deleteDatabase();
+        PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        adapter.close();
+
+        // Emulate turned off gpodder.net support in SyncService
+        SharedPreferences spref = mock(SharedPreferences.class);
+        when(spref.getString(eq("prefGpodnetHostname"), anyString())).thenReturn("gpodder.net");
+        Application app = mock(Application.class);
+        when(app.getSharedPreferences(anyString(), anyInt())).thenReturn(spref);
+        ClientConfig.applicationCallbacks = mock(ApplicationCallbacks.class);
+        when(ClientConfig.applicationCallbacks.getApplicationInstance()).thenReturn(app);
+
+        localFeedDir1 = new File("sampledata/local-feed1");
+        localFeedDir2 = new File("sampledata/local-feed2");
+        List<File> files = new ArrayList<>();
+        Collections.addAll(files, localFeedDir1.listFiles());
+        Collections.addAll(files, localFeedDir2.listFiles());
+        for (File file : files) {
+            DataSource ds = DataSource.toDataSource(context, Uri.fromFile(file));
+            ShadowMediaMetadataRetriever.addMetadata(ds, MediaMetadataRetriever.METADATA_KEY_DURATION, "10");
+            ShadowMediaMetadataRetriever.addMetadata(ds, MediaMetadataRetriever.METADATA_KEY_TITLE, file.getName());
+        }
+        shadowOf(MimeTypeMap.getSingleton()).addExtensionMimeTypMapping("mp3", "audio/mp3");
+    }
+
+    @SuppressWarnings("JavaReflectionMemberAccess")
+    @After
+    public void tearDown() throws Exception {
+        // Workaround for Robolectric issue in ShadowSQLiteConnection: IllegalStateException: Illegal connection pointer
```

Comment:
This looks pretty hacky. What about adding a method directly to `PodDbAdapter`? Something like `PodDbAdapter.tearDownTests()`?

Reply:
You're right, it's ugly code. :sob:
An approach like `PodDbAdapter.tearDownTests()` couldn't cover all needs without the help of Mockito as the field `SQLiteOpenHelper.mContext` is declared as final, but it has to be set to null after each test to avoid the "Illegal connection pointer" error when using Robolectric. For more information see the discussion in https://github.com/robolectric/robolectric/issues/1890.

Reply:
Hmm, okay then I would suggest to at least hide the ugliness somewhere in another class (basically like `EspressoTestUtils`, just for `RobolectricTestUtils`)

Reply:
Sorry, my answer is wrong. The field `SQLiteOpenHelper.mContext` is not touched, instead the field `SQLiteOpenHelper.mDatabase` has to be reset which is not final. Finally, I could introduce the method `PodDbAdapter.tearDownTests()` as suggested and replace the hacky reflection code. :grin: